libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.h

```diff
@@ -47,4 +50,6 @@ class AP_BattMonitor_UAVCAN : public AP_BattMonitor_Backend
 
     AP_UAVCAN* _ap_uavcan;
     uint8_t _node_id;
+    mutable uint8_t _soc;
```

Comment:
This would be the first use of `mutable` in the ArduPilot codebase.  That would require some discussion - and its not required if you don't assign into _soc_pct (see proposed modification)